src/lib/bcmp/bcmp_neighbors.cpp

```diff
@@ -67,14 +66,19 @@ void bcmp_neighbor_foreach(neighbor_callback_t cb) {
 }
 
 static void _neighbor_check(bm_neighbor_t *neighbor) {
-  if (neighbor->online &&
-      !timeRemainingTicks(neighbor->last_heartbeat_ticks,
-                          pdMS_TO_TICKS(2 * neighbor->heartbeat_period_s * 1000))) {
-    printf("🏚  Neighbor offline :'( %016" PRIx64 "\n", neighbor->node_id);
-    if (NEIGHBOR_DISCOVERY_CB) {
-      NEIGHBOR_DISCOVERY_CB(false, neighbor);
+  int32_t time_remaining = 0;
+  int32_t time_start = (int32_t)neighbor->last_heartbeat_ticks;
+  int32_t time_end = 0;
+  if (neighbor->online) {
+    time_end = (int32_t)bm_ms_to_ticks(2 * neighbor->heartbeat_period_s * 1000);
+    time_remaining = time_start + time_end - (int32_t)bm_get_tick_count();
+    if (time_remaining <= 0) {
```

Comment:
This seems very easy to subtly mess up. Time rollover has been the source of multiple serious bugs for us. The casting to `int32_t` raises red flags for me. Tell me more about the rationale for changing it?

Reply:
This was a function that was tied to external API outside of bcmp/bristlemouth. I decided to carry over the logic to here. Maybe the better choice is to place it in `util.h` in `bm_core`?

Reply:
I adressed this by placing the API it in `bm_common` and keeping the function call logic the same.